### PR TITLE
Review & Enhance Service Tests for Contact endpoints

### PR DESCRIPTION
**Pull request by GPaoloni**
## Description
Reworked `hrm-service/service-tests/contacts.test.js` tests to use (mostly) `jest-each` and added test cases (for query params, single search param, date filters, among others). Removed `CaseAudits` tests, instead checking against `Audits` table (not really necessary, maybe we want to remove this).

### Checklist
- [x] [Corresponding issue has been opened](https://bugs.benetech.org/browse/CHI-989)
- [x] New tests added

### Verification steps
- Check that CI is still passing locally and remotely after the changes.
- Manual review to make sure test cases makes sense (:sweat_smile: )?

### CODE REVIEW

File: hrm-service/service-tests/contacts.test.js [hunk 1]
```diff
@@ -169,11 +298,100 @@ describe('/contacts route', () => {
...
+      await Contact.destroy(query);
+
+      // Create some contacts to work with
+      const contact = { ...contact1 };
+
+      const withQueueName = { ...contact, queueName };
+
+      const oneHourBefore = {
+        ...withQueueName,
+        timeOfContact: new Date(new Date().getTime() - 1000 * 60 * 60).toISOString(), // one hour before
```
Comment (stephenhand) [inline hrm-service/service-tests/contacts.test.js:315]: `subHours(new Date(), 1).toisoString()` might be more readable than hand rolling it. `date-fns` has functions for most basic date operations, and it's already a dependency so might as well use it!

**Review by stephenhand — COMMENTED** [review]
Looks good. One thing, are we specifically testing that CSAM reports are retrieved with contacts in the GET and search endpoints? That's one of the things I'd be most concerned about breaking in a refactor of the Contacts DB access - joins are easy to inadvertently break

File: hrm-service/service-tests/contacts.test.js (hunk 1 above)
Comment (GPaoloni) [reply hrm-service/service-tests/contacts.test.js]: Thanks for the suggestion! :raised_hands:

**Comment by GPaoloni** [comment]
@stephenhand Both suggestions addressed in https://github.com/techmatters/hrm/pull/180/commits/ed96e455d746f49474c4ae7a392fbc0401e46868

**Review by stephenhand — APPROVED** [review]

**Pull request merged**